File: solutions/ruby/anagram/1/anagram.rb
class Anagram
  def initialize(word)
    @word = word.downcase
    @sorted_word = @word.chars.sort
  end

  def match(candidates)
    candidates.select do |candidate|
      cand_lower = candidate.downcase
      cand_lower != @word && cand_lower.chars.sort == @sorted_word
    end
  end
end

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    word_lower = word.lower()
    word_sorted = sorted(word_lower)
    res = []
    for cand in candidates:
        cand_lower = cand.lower()
        if cand_lower != word_lower and sorted(cand_lower) == word_sorted:
            res.append(cand)
    return res
